server/revision/revision.controller.js

```diff
@@ -1,14 +1,24 @@
 'use strict';
 
-const { Revision, User } = require('../shared/database');
+const {
+  Activity,
+  ContentElement,
+  Revision,
+  Sequelize: { Op },
+  User
+} = require('../shared/database');
+const { getOutlineLevels } = require('../../config/shared/activities');
+const map = require('lodash/map');
 const { resolveStatics } = require('../shared/storage/helpers');
 
-function index({ repository, query }, res) {
-  const { limit, offset, entity, entityId } = query;
+async function index({ repository, query }, res) {
+  const { limit, offset, entity, entityId, descendants } = query;
   const where = { repositoryId: repository.id };
   if (entity) {
     where.entity = entity;
     where.state = { id: entityId };
+  } else if (descendants) {
+    where[Op.or] = await buildDescendantsCondition(entityId, repository);
```

Comment:
If I am correctly understanding the usage, we should just rely on timestamps here (not revisions).  This can be very load intensive. If we want to list all changes we should do paginated version upon item hover (not immediately)

Reply:
We are interested in all activity and content element revisions within specific outline activity so this condition is needed? I wasn’t aware of the fact that content containers are available client side before opening editor so I'll move it there.